docs/usage/config-presets.md

```diff
@@ -232,6 +232,38 @@ Parameters are supported similar to other methods:
 }
 ```
 
+## Templating presets
+
+You can use [Handlebars](https://handlebarsjs.com/) templates to be flexible with your presets.
+This can be handy when you want to include presets conditionally.
+
+<!-- prettier-ignore -->
+!!! note
+    The template only supports a small subset of options, but you can extend them via `customEnvVariables`.
+
+Read the [templates](./templates.md) section to learn more.
+
+### Example use-case
+
+The following example shows a self-hosted Renovate configuration for a company called "Renovators".
+The company is split into multiple organizations:
+
+- `foo`
+- `bar`
+- `baz`
+
+Each organization provides its opinionated shareable preset.
```

Comment:
Isn't this use case satisfied by https://docs.renovatebot.com/self-hosted-configuration/#inheritconfig ?

Reply:
Hmm... I didn't know about this. So we probably would need to find another example, since `inheritConfig` does not cover the use-case of adding a dynamic reference ([see PR description](https://github.com/renovatebot/renovate/pull/27955#issue-2188579889)).

Reply:
@rarkins how about this use case:
Having a central Renovate preset (default.json) located in a repository `renovate/presets`. But actually before blindly pushing changes to main, it would be nice to validate the changes. This can be achieved by using templating:

```javascript
// config.js
module.exports = {
  customEnvVariables: {
    GITLAB_REF: process.env.CI_COMMIT_REF_NAME || "main",
  },
  extends: ["local>renovate/presets#{{ env.GITLAB_REF }}"]
}
```

I found an older discussion, which could benefit from this feature:
https://github.com/renovatebot/renovate/discussions/16108

Reply:
Updated the docs, please have another look.